### ISSUE TITLE
[Agent]: Windows agent shows up unhealthy  with metricbeat error logs in Agents tab when 'linux' Integration is used in agent policy.

### ISSUE THREAD

**Issue by dikshachauhan-qasource** [elastic/beats#24105]
**Build**: 7.11.1 snapshot staging build

**On installing agent using a policy having on windows and Linux integration only on Windows x64 bit, we observed that just after installation agent goes to **unhealthy** state. on Verifying the logs we **observed below error**.**

Template: windows10x64 testsigning ON

**Metric beat failure error logs in Agent tab:**
```
17:25:07.328
elastic_agent
[elastic_agent][error] 2021-02-18T06:55:06-05:00: type: 'ERROR': sub_type: 'FAILED' message: Application: metricbeat--7.11.1-SNAPSHOT[126b0a70-71e0-11eb-8a6d-650e4bf6beba]: State changed to FAILED: 1 error: 5 errors: Error creating runner from config: 1 error: error connecting to dbus: dial tcp 127.0.0.1:12434: connectex: No connection could be made because the target machine actively refused it.; Error creating runner from config: 1 error: metricset 'system/socket' not found; Error creating runner from config: 1 error: metricset 'system/users' not found; Error creating runner from config: 1 error: metricset 'system/entropy' not found; Error creating runner from config: 1 error: could not read \proc: CreateFile \proc: The system cannot find the file specified.
```

**Screenshot:**
![image](https://user-images.githubusercontent.com/72375798/108355322-ae0c0a80-7210-11eb-9ec3-f51a8a380a06.png)

**Build details:**
```
BUILD 37947
COMMIT b8ce8c202dff437413cdd01d5ba956bfa41bb8af
artifact link: https://artifacts-api.elastic.co/v1/search/7.11.1-SNAPSHOT
```

**Comment by dikshachauhan-qasource** [comment]
Please review @manishgupta-qasource

**Comment by manishgupta-qasource** [comment]
Reviewed & Assigned to @EricDavisX

**Comment by EricDavisX** [comment]
we were already workin on 'guards' in Metricbeat and Filebeat packages for when all OSes are used in same policy, I think this is the same bug - we can test it when we get the next version in to staging to test.

**Comment by fearful-symmetry** [comment]
Yah, we need to know what version of the system integration this is.

**Comment by EricDavisX** [comment]
@fearful-symmetry it is 0.10.7

**Comment by fearful-symmetry** [comment]
Okay. 0.10.7 is a tad old at this point. There's a newer version in staging at least, but the most recent is 0.11.

**Comment by EricDavisX** [comment]
That's totally fair  :)  And I might actually be wrong it may have been 0.10.7 since it is a 7.11 snapshot...  the team had several systems they were working yesterday, and I am guessing since it was ommited.

Regardless, we are trying to manage tests over the 3 different 'current' versions of System Integration based on what is available differently (as intended) for the 4 relating dev lines we focus on (7.10.x, 7.11.x, 7.12.0 and 8.x) which are all different (unsure about 7.12 we just started testing there).  If you would prefer we re-test this with 0.10.9 that sounds good to me, but the ticket does reflect what a shipped version of Fleet / Agent are experiencing with a shipped or soon-to-ship version of Integration, so it can help prioritize the work (particularly if severe, like the guards work being done).

Just for notes:
version - stage of integration - version of stack
system-0.9.3	production - 0.9.x is tied to 7.10.x
system-0.10.7	production - 0.10.x is tied to 7.11.x
system-0.10.9	staging - 0.10.x is tied to 7.11.x - we are hoping to finishing tests and promote this imminently 
system-0.11.0	snapshot - 0.11.x is tied to 8.x and we have not evaluated it in detail.  

If you are done with your own tests and want to request a test run of 8.x with System (and other new) packages please let us know, we're happy to run it.

For now, let us push what content we can to staging and the team can test it, we just need to set expectations otherwise it is hard to know if a problem seen is new or not.

**Comment by fearful-symmetry** [comment]
Okay, sorry, confused myself here. The actual changes that impact this bug are in the `linux` module. The fixes should be in `0.3.8`, which I just merged to `SNAPSHOT`. I'll put in the PR for staging now.

**Comment by EricDavisX** [comment]
not a problem at all, lol - all of a sudden this week it got to be a lot more to try to manage. I'm very excited to see what the package-storage updates to support tracking versions more automatedly may end up looking like (from https://github.com/elastic/package-storage/pull/939/files)

**Comment by dikshachauhan-qasource** [comment]
Hi @EricDavisX cc @fearful-symmetry 

Thanks for the update. Please let us know whenever Linux package version 3.8 is out . We will retest the above issue and will close it accordingly. Currently, 7.11.1 snapshot build is having below mentioned package versions.:
System: 10.9
Windows: 3.1
Linux: 3.7

Thanks
QAS

**Comment by fearful-symmetry** [comment]
So, `0.8.3` of Linux should be available in staging now. I think.

**Comment by dikshachauhan-qasource** [comment]
Hi @EricDavisX cc @fearful-symmetry 

We have revalidated this issue on 7.11.1 snapshot Kibana cloud build having Linux package version 3.8 and found issue as **Fixed**.

- Now, Windows agent shows up healthy with metricbeat logs as Running  in Agents tab when 'linux' Integration is used in agent policy. 

**Screenshot:**
![image](https://user-images.githubusercontent.com/72375798/108810022-2b989780-75d0-11eb-80cc-5f5102e5bd4b.png)

**Build details:**
```
BUILD 37947
COMMIT b8ce8c202dff437413cdd01d5ba956bfa41bb8af
Artifact link: https://snapshots.elastic.co/7.11.1-16d3597f/downloads/beats/elastic-agent/elastic-agent-7.11.1-SNAPSHOT-windows-x86_64.zip
```

Hence, closing this issue.

Thanks
QAS

**Comment by EricDavisX** [comment]
@dikshachauhan-qasource thanks.

@fearful-symmetry let us do the PR to push Linux + Windows to prod and get them rolled out.  Diksha can do a quick re-test after the roll out of storage and report back.  Thanks everyone for the efforts here.

**Comment by dikshachauhan-qasource** [comment]
Hi @EricDavisX cc @fearful-symmetry 

We have revalidated windows package 4.1 on 7.11.1 snapshot Stack environment and found it working fine.

Observations:
- Agent was in healthy state.
- Windows package installed successfully[Agent policy having windows, Linux and System integrations]
- Data streams showed up below windows logs and metrics. [Except forwarded and sysmon logs]
- Logs were available on discover tab

![image](https://user-images.githubusercontent.com/72375798/109152033-f2f3ec00-7790-11eb-8ed7-5d0cac010281.png)

@fearful-symmetry Logs entry for windows.powershell and windows.powershell_operational were added up to datastream only once and not updated even we attempted perform some actions. Could you please confirm expected behavior.


**Build details:**
BUILD 37947
COMMIT b8ce8c202dff437413cdd01d5ba956bfa41bb8af
Artifact link: https://snapshots.elastic.co/7.11.1-16d3597f/downloads/beats/elastic-agent/elastic-agent-7.11.1-SNAPSHOT-windows-x86_64.zip

Thanks
QAS

**Comment by EricDavisX** [comment]
I assume that is with the default Integration policy configuration settings.

This is a specific concern we can log to a new issue if needed / appropriate.

For now with this confirmation it is enough for us to publish it to production, I'll work with Alex to do this (today hopefully) and we can re-test.  @ph we're in good shape now with with 3 Integrations (System, Windows, Linux).  And the team will look at System 11 package separately, against 8.0 after the 7.12 cycle is done, fyi.

**Comment by dikshachauhan-qasource** [comment]
Hi @EricDavisX 

Thanks for the feedback and will and will log the query raised above separately.

CC @fearful-symmetry 

Thanks
QAS

**Comment by dikshachauhan-qasource** [comment]
Hi @EricDavisX 

The query we have raised is already reported at [#669](https://github.com/elastic/integrations/issues/669).

We have updated ticket with our observations. Please let us know if anything is required from our side.

Thanks
QAS
